### ISSUE TITLE
Todo list

### ISSUE THREAD

**Issue by bas-rustenburg** [choderalab/openmm-constph#20]
Brainstorming things to do before we can start using this code more reliably.
Tagging @jchodera @gregoryross, feel free to add points by editing (assuming you can) or replying. 

- [ ] Clean up the repository
  - [ ] docstrings
  - [ ] unused files
  - [ ] directory structure
- [ ] setup.py & conda package
  - using the usual devtools setup
- [ ] Add tests and continuous integration (Travis)

- [ ] Validate that calibrations with SAMS works well enough
  - [ ] Investigate convergence of ref. free energies.
    - Implict vs explict solvent calibrations
    - Effect of counterions in explicit calibration energies
    - Variance estimation

  - We need to implement some functions before we can use calibration:
    - [ ] set up easy calibration at given pKa for amino acids
    - [ ] set up ligand system (solvation?) and calibrate automatically as well

- [ ] Maintain charge neutrality by inserting/deleting counter ions (replacing water)
  - Remove dependency on pre-existing ions
  - add user specified ions instead

- [ ] Small molecule input files cpin/prmtop/ffxml
  - Currently working on an ffxml format that takes input from epik, relies on openmoltools
    - [ ] Needs work on openmoltools for implicit solvent support
   
- [ ]  Implement constant-counterions
- [ ]  Set up test systems

Lower priority/ extra features:
- [ ] pKa biasing
- [ ] Selection of residue pairs for simultaneous (de)protonation
- [ ] Getting rid of cpin files
  - [ ] Protein constant-pH ffxml?

**Comment by jchodera** [comment]
Thanks!  This is great.

I'd can help overhaul the NCMC and counterion insertion/deletion code.  I should have time to do this Tue.
